xapian-letor/ranker/svmranker.cc

```diff
@@ -45,6 +45,20 @@ Proceedings of the eighth ACM SIGKDD international conference on Knowledge disco
 using namespace std;
 using namespace Xapian;
 
+static void
+clear_svm_problem(svm_problem problem, int n)
+{
+    delete [] problem.y;
+    problem.y = NULL;
+    for (int i = 0; i < n;i++)
+    {
```

Comment:
> I don't know how did i pass xapian-check-patch test on travis C.I after having i++

The script isn't infallible - it's essentially a lot of regular expression checks, and certainly not a full C++ parser (to use a C++ parser, I think we'd need to work from the source file rather than just a patch, as the patch has limited context - I've been wondering if that would be a better approach, but for now we have what we have).

Anyway. the current version of this particular check was defeated by the missing space before `i++`.  I'll push a fix for that, and also add a check so it catches the missing space after `;`.

> I always do rush in pushing commits

Don't rush to commit!  That's really the root of the problem here.

Taking your time checking things over takes a few minutes for a patch of this size.  Pushing and seeing the CI patch check fail takes longer, and also is really abusing the CI resources which are provided for free.  An extra review cycle by the maintainer takes hours or days in this context, and for many projects it can be weeks or months - I'm not exaggerating here, some examples from looking at my github PR history:

* 7 months: https://github.com/formorer/debian-indexer-xapian/pull/4
* 8 months: https://github.com/therion/therion/pull/1
* 18 months: https://github.com/dagwieers/unoconv/pull/174
* 22 months: https://github.com/madler/zlib/pull/85

Luckily I didn't mess up on any of those!

Admittedly those are some of the outliers from a fairly large set, but weeks is certainly common.

Most maintainers like bugs in their code to be fixed for them and would love more developers to contribute to their projects, so there's a clear reason to review and merge patches.  But in many cases, they're struggling to find time to review all the patches promptly.  If you submit clean patches they can just merge right away you're likely to get reviewed more promptly than people submitting scrappy patches which need many iterations.  If you can show you're a safe pair of hands, you'll often get the ability to just push changes directly without a review up front.  But if you don't address all the comments before pushing a new version, or you make the same mistakes time after time, you'll tend to just drift down the review pile.